syntax = "proto3";

package masterworker;

service MapReduceMaster {
  rpc MapReduceTask(MapReduceQuery) returns (MapReduceReply) {}
}

message ShardFile {
  string filename = 1;
  int32 offset_start = 2;
  int32 offset_end = 3;
}

message MapReduceQuery {
  string user_id = 1;
  repeated ShardFile shard = 2;
  int32 n_output_files = 3;
  string output_directory = 4;
  int32 mapping = 5;
  string temp_file = 6;
}

message MapReduceReply {
  repeated ShardFile output_files = 1;
  int32 complete = 2;
}
